File: nanomsg/01_simple_sender_receiver/sender.c
#include <stdio.h>
#include <string.h>
#include <unistd.h>
#include <nanomsg/nn.h>
#include <nanomsg/pipeline.h>

const char *URL = "ipc:///tmp/example1";

void sender(const char *url, const char *message)
{
    int message_size = strlen(message) + 1;
    int socket;
    int endpoint;
    int bytes;

    socket = nn_socket(AF_SP, NN_PUSH);
    puts("Socket created");

    endpoint = nn_connect(socket, url);
    puts("Remote endpoint added to the socket");

    printf("Sending message '%s'\n", message);
    bytes = nn_send(socket, message, message_size, 0);

    printf("Message with length %d bytes sent, flushing", bytes);
    sleep(1);
    puts("Done");

    nn_shutdown(socket, endpoint);
}

int main(const int argc, const char **argv)
{
    sender(URL, "Hello");
    sender(URL, "world");
    sender(URL, "!");

    return 0;
}

File: nanomsg-python/01_sender_receiver/sender.py
# vim: set fileencoding=utf-8

from nanomsg import PUSH, Socket

URL = "ipc:///tmp/example1"

socket = Socket(PUSH)
print("Socket created")

socket.connect(URL)
print("Connected to URL {}".format(URL))

socket.send("Hello world!")
print("Message has been sent")

socket.close()
print("Socket closed")
